### DISCUSSION TITLE
ltree

### DISCUSSION THREAD

**Discussion by cemoktra** [SeaQL/sea-orm#652] [General]
We just played around with ltrees. `sea-orm-cli` currently generates an empty enum. Maybe for the special ltree case it shoud generate something like

```
#[derive(Debug, Clone, PartialEq, EnumIter, DeriveActiveEnum)]
#[sea_orm(rs_type = "String", db_type = "Enum", enum_name = "ltree")]
pub enum Ltree {
    Value(String)
}
```

Currently this ends up in `error: Missing macro attribute, either `string_value` or `num_value` should be specified`, so there would need to be a way to extract the string out of the Value variant.

**Comment by billy1624** [comment]
Hey @cemoktra, I guess you need to implement `ActiveEnum` manually instead of deriving `DeriveActiveEnum`.
https://www.sea-ql.org/SeaORM/docs/generate-entity/enumeration#manual-implementation

**Comment by cemoktra** [comment]
Ok also possible, and what about mapping ltree to TEXT?

**Comment by cemoktra** [comment]
But I think I'll provide some changes for ltree

**Comment by billy1624** [comment]
I'm not sure how `ltree` works.

But for `ActiveEnum`,
- on insert, it will generate SQL `'enum_value' AS enum_type`
- on select, it will generate SQL `enum_column AS text`

**Comment by cemoktra** [comment]
I started with the basic work on sqlx: https://github.com/launchbadge/sqlx/pull/1696

**Comment by billy1624** [comment]
Thanks!! @cemoktra
Let me know if you need any help on it!

**Comment by cemoktra** [comment]
Yeah maybe. Currently sea-query does not depend on sqlx itself, but in order to use my new type it will

**Comment by cemoktra** [comment]
Do you have any opinion on how to integrate this into seaql?

**Comment by billy1624** [comment]
I think we can treat ltree as a simple string? Then, we don't need any additional SQLx dependency.
https://github.com/launchbadge/sqlx/pull/1696#discussion_r804343242

**Comment by cemoktra** [comment]
Currently i get `thread 'test::test' panicked at 'not implemented', sea-orm/src/driver/sqlx_postgres.rs:8:1` when trying to filter for an LTree with .eq, also .contains wants a String instead of PgLTree

```
.filter(test::Column::Ltree.eq(ltree))
```

Just doing a query works

**Comment by cemoktra** [comment]
Same for insert

**Comment by cemoktra** [comment]
Ok found the missing piece, sorry

**Comment by tyt2y3** [comment]
This is sad but I think our general response to "support a particular Postgres feature" will be "we will gladly accept PR" for now
